### ISSUE TITLE
IDE0130 message "Namespace 'Foo' does not match folder structure, expected 'Foo'"

### ISSUE THREAD

**Issue by manideep418** [dotnet/roslyn#55550]
Hello team,

There are changes to the docker file today which are causing the build failures using the below command

RUN dotnet build "Cardlytics.Api.CampaignBuild.Web.csproj" -c Release -o /app /p:TreatWarningsAsErrors=true /warnaserror

sample error: error IDE0130: Namespace "abcXYZ" does not match folder structure, expected "abcXYZ"

**Comment by adamralph** [comment]
I'm seeing a slight variant of this - https://github.com/adamralph/minver/runs/3386153426?check_suite_focus=true#step:6:11

> warning IDE0130: Namespace "MinVerTests.Infra" does not match folder structure, expected "MinVerTests.Infra."

It only happens on macOS. It does not happen on Linux or Windows.

**Comment by adamralph** [comment]
Also happens in Visual Studio:

![image](https://user-images.githubusercontent.com/677704/130318234-30d706f3-625c-487a-b8b5-d2b7b47b7b66.png)

**Comment by adamralph** [comment]
![image](https://user-images.githubusercontent.com/677704/130318558-d49db12a-a7ff-40aa-91c8-338fa2a4035b.png)

**Comment by jcouv** [comment]
Moved to area IDE since relates to error `IDE0130`

**Comment by ryzngard** [comment]
This should be fixed by https://github.com/dotnet/roslyn/pull/55424/files#diff-e0e4d96e8b60f4b8198b8b16e11550b47de27f191802beb6c61379f7b44c6b39R30 

The fix is in Preview 4 for VS 2022

**Comment by adamralph** [comment]
@ryzngard is the fix already in a .NET 6 SDK release?

**Comment by ryzngard** [comment]
I'm honestly not 100% sure on the shipping date relations between Roslyn CodeStyle layer and SDK. @jmarolf do you know?

**Comment by adamralph** [comment]
@ryzngard no worries, this seems to be a universal and eternal guessing game for all the things that feed into the SDK.

**Comment by ryzngard** [comment]
I'm going to close this as fixed, but would still like clarity on where this fits in with the SDK. @jmarolf or @mavasani do you know?

**Comment by jmarolf** [comment]
running the following command:
```cmd
> git tag --contains 00a15e7ce4a266189e896826833c1dedf32dcf2f
Visual-Studio-2022-Version-17.0-Preview-4
Visual-Studio-2022-Version-17.0-Preview-4.1
Visual-Studio-2022-Version-17.0-Preview-5
```
This fix shipped in VS 2022 Preview 4 which corresponds to .NET 6 RC1. If you have .NET 6 RC1 or newer it should contain this fix.

**Comment by jmarolf** [comment]
> Where is the VS to SDK mapping shown?

The unfortunate reality is Visual Studio previews and SDK previews and their alignment cannot be shown. The SDK preview release cycle is public (a new preview releases each month from Jan to Oct each year). But the Visual Studio one changes release to release and I cannot share it publicly. behind the scenes the SDK team works tirelessly to try and get the latest SDK preview that is available into Visual Studio previews but the only time things ever truly align is when VS and .NET RTM.

I know that VS 2022 Preview 4 corresponds to .NET 6 RC1 because I went back and checked our blog posts to ensure we were touting the SDK RC1 release in VS 2022 Preview 4.

> BTW - assuming that this package feeds into the SDK and in turn, the SDK feeds into VS, would it be better to tag the SDK version rather than the VS version?

Not a bad idea. The reality is that the .NET SDK and Visual Studio are separate products with separate release cycles. We should probably do the work to tag both the SDK and Visual Studio release in this repo.

**Comment by adamralph** [comment]
FWIW, I can confirm that this problem no longer exists in 6.0.100 (RTM).

**Comment by lennartb-** [comment]
We're still seeing this issue on our CI. How can we check that we have a fixed version installed? `dotnet --info` returns

```
PS C:\Windows\system32> dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.101\

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  2.1.801 [C:\Program Files\dotnet\sdk]
  5.0.404 [C:\Program Files\dotnet\sdk]
  6.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by ryzngard** [comment]
@lennartb- I was unable to reproduce this manually in 17.0.5. The version listed should have the fix.

**Comment by lennartb-** [comment]
I'll check with 17.05 again, we deferred upgrade to this version, since there are no release notes available yet.

**Comment by ryzngard** [comment]
@lennartb- it should be the same as 17.0.4 with regards to this fix, I just didn't have a 17.0.4 build readily available to test
